### PR TITLE
Adds file system manager

### PR DESCRIPTION
**Pull request by devstepien**
## Description

This PR:
- Adds File System manager (to read and write messages from the file).
- There is a prototype for the group manager, but we can change it later.

The approach being taken here is: for each group we will have one file that will maintain the messages. The read and write for those files that need to be protected by a mutex. Other groups should be able to persist/read the messages separately (group B should not block group A).

Saves the messages as a CSV

### TODO:
- Think in a way to protect the write to the group files.

### CODE REVIEW

File: Instant-Messenger/include/util/file_system_manager.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#define FILE_SEPARATOR ","
+#define PATH_SEPARATOR "/"
+
+using namespace std;
+
+using std::string;
+
+namespace filesystemmanager {
+
+using namespace message;
```
Comment (cassianojaeger) [inline Instant-Messenger/include/util/file_system_manager.hpp]: Is there any difference from declaring using name space inside a namespace declaration? I mean, ServerMessageServer declares its using namespaces outside the namespace declaration for that specific file.

File: Instant-Messenger/include/server/server_group_manager.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include <string>
```
Comment (cassianojaeger) [inline Instant-Messenger/include/server/server_group_manager.h]: Should not this be a .hpp file?

Comment (devstepien) [reply Instant-Messenger/include/server/server_group_manager.h]: yeap, our include files should be hpp, I'll changes this

File: Instant-Messenger/include/util/file_system_manager.hpp (hunk 1 above)
Comment (devstepien) [reply Instant-Messenger/include/util/file_system_manager.hpp]: > As a rule of thumb, using declarations should only be used after all includes in a .cpp file. There's also an item on this exact issue in the book "C++ Coding Standards" by Sutter and Alexandrescu (Item 59). Here's a quote: "But here's the common trap: Many people think that using declarations issued at namespace level (...) are safe. They are not. They are at least as dangerous, and in a subtler and more insidious way."

https://stackoverflow.com/questions/6175705/scope-of-using-declaration-within-a-namespace

Comment (devstepien) [reply Instant-Messenger/include/util/file_system_manager.hpp]: I'll keep it out of the namespace.

**Review by Andy9822 — APPROVED** [review]

**Pull request merged**